feat: Client Reddit Files

Freytes opened the pull request:
# @ai16z/client-reddit

A powerful Reddit client plugin for the Eliza framework that enables AI agents to interact with Reddit through posting, commenting, and voting capabilities.

## Features

* Full Reddit API integration via Snoowrap wrapper
* Create and submit posts to multiple subreddits
* Comment on existing posts and replies
* Vote on posts and comments programmatically
* Automated posting with configurable intervals
* Built-in rate limiting and error handling
* Dry run mode for testing
* TypeScript support out of the box

## Installation

```bash
npm install @ai16z/client-reddit
```

Or using yarn:

```bash
yarn add @ai16z/client-reddit
```

## Configuration

### Required Environment Variables

```env
REDDIT_CLIENT_ID=your_client_id
REDDIT_CLIENT_SECRET=your_client_secret
REDDIT_REFRESH_TOKEN=your_refresh_token
REDDIT_USER_AGENT=your_user_agent
REDDIT_SUBREDDITS=subreddit1,subreddit2,subreddit3
```

### Optional Configuration

```env
REDDIT_AUTO_POST=true           # Enable automated posting
POST_INTERVAL_MIN=90           # Minimum time between posts (minutes)
POST_INTERVAL_MAX=180          # Maximum time between posts (minutes)
POST_IMMEDIATELY=false         # Post on startup
REDDIT_DRY_RUN=false          # Run without making actual API calls
```

## Usage

### As an Eliza Plugin

```typescript
import { redditPlugin } from '@ai16z/client-reddit';
import { Eliza } from '@ai16z/eliza';

const eliza = new Eliza();
eliza.use(redditPlugin);

// The plugin will automatically initialize with your environment variables
```

### Direct Usage

```typescript
import { RedditClient } from '@ai16z/client-reddit';

const client = new RedditClient(runtime);
await client.start();

// Submit a post
await client.submitPost('subreddit', 'Title', 'Content');

// Create a comment
await client.createComment('t3_postId', 'This is a comment');

// Vote on content
await client.vote('t3_postId', 1); // 1 for upvote, -1 for downvote
```

## Available Actions

### Create Post

```typescript
runtime.execute('CREATE_REDDIT_POST', {
  content: {
    subreddit: 'test',
    title: 'My First Post',
    text: 'This is the content of my post'
  }
});
```

### Create Comment

```typescript
runtime.execute('CREATE_REDDIT_COMMENT', {
  postId: 't3_abc123',
  content: 'This is my comment'
});
```

### Vote

```typescript
runtime.execute('REDDIT_VOTE', {
  targetId: 't3_abc123',
  direction: 1  // 1 for upvote, -1 for downvote
});
```

## Error Handling

The client includes built-in error handling for common Reddit API issues:

* Rate limiting
* Invalid credentials
* Network errors
* Subreddit posting restrictions
* Content filters

Errors are logged and can be caught using standard try/catch blocks.

Powershell Script required to make an OAuth Key:

```
# Reddit App Credentials
$CLIENT_ID = "xxx"
$CLIENT_SECRET = "xxx"
$REDIRECT_URI = "http://localhost:8080/callback"

# Generate random state
$state = -join ((65..90) + (97..122) | Get-Random -Count 16 | % {[char]$_})

# Define scopes
$SCOPES = "identity submit edit vote read"

# Create authorization URL
$authUrl = "https://www.reddit.com/api/v1/authorize?" +
           "client_id=$CLIENT_ID&" +
           "response_type=code&" +
           "state=$state&" +
           "redirect_uri=$([Uri]::EscapeDataString($REDIRECT_URI))&" +
           "duration=permanent&" +
           "scope=$([Uri]::EscapeDataString($SCOPES))"

Write-Host "Visit this URL in your browser to authorize the app:"
Write-Host $authUrl
Write-Host "`nAfter authorization, you'll be redirected to a URL like:"
Write-Host "http://localhost:8080/callback?state=xyz&code=ABC123..."
Write-Host "`nPaste ONLY the code value (the part after 'code=' and before any '#' or '&'):"
$code = Read-Host "Enter the code"

# Exchange code for tokens
$base64Auth = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes("${CLIENT_ID}:${CLIENT_SECRET}"))
$headers = @{
    "Authorization" = "Basic $base64Auth"
    "Content-Type" = "application/x-www-form-urlencoded"
}

$body = "grant_type=authorization_code&code=$code&redirect_uri=$([Uri]::EscapeDataString($REDIRECT_URI))"

try {
    $response = Invoke-RestMethod `
        -Uri "https://www.reddit.com/api/v1/access_token" `
        -Method Post `
        -Headers $headers `
        -Body $body

    Write-Host "`nRefresh Token: $($response.refresh_token)"
    Write-Host "Access Token: $($response.access_token)"
} catch {
    Write-Host "Error: $($_.Exception.Message)"
    if ($_.Exception.Response) {
        $result = $_.Exception.Response.GetResponseStream()
        $reader = New-Object System.IO.StreamReader($result)
        $reader.BaseStream.Position = 0
        $reader.DiscardBufferedData()
        $responseBody = $reader.ReadToEnd()
        Write-Host "Response: $responseBody"
    }
}
```

Please perform the following prior to generating an OAuth Key:

1. Create an account for your agent on: Reddit.com
2. Login using the newly created account.
3. Create a script based application:
```
script | Script for personal use. Will only have access to the developers accounts
redirect uri | http://localhost:8080/callback
```
4. Use the powershell script logged in as the agent, to generate the OAuth Key.
5. Place the key as the: REDDIT_REFRESH_TOKEN

odilitime commented:
needs to include the agent/ wiring (package.json src/index.ts)

Freytes commented:
@odilitime Updated the requested files.

shakkernerd commented:
You need to rebase your main, seems like some changes are missing in your fork.

odilitime commented:
Tried to resolve conflicts but looks like `src/agent/index.ts` is missing some updates from develop, we can't merge until those regressions are fixed.

odilitime commented on agent/src/index.ts:
```diff
@@ -78,8 +79,7 @@ export const wait = (minTime: number = 1000, maxTime: number = 3000) => {
     elizaLogger.debug(`Fetching ${url}`);
-    // Disabled to avoid disclosure of sensitive information such as API keys
-    // elizaLogger.debug(JSON.stringify(options, null, 2));
+    elizaLogger.debug(JSON.stringify(options, null, 2));
```
this shouldn't be enabled

odilitime commented on agent/src/index.ts:
all these regressions needs to be fixed

houssemmechri commented on agent/src/index.ts:
```diff
@@ -7,6 +7,7 @@ import { LensAgentClient } from "@elizaos/client-lens";
 import { SlackClientInterface } from "@elizaos/client-slack";
 import { TelegramClientInterface } from "@elizaos/client-telegram";
 import { TwitterClientInterface } from "@elizaos/client-twitter";
+import { RedditClientInterface } from "@elizaos/plugin-reddit";
```
this should be client-reddit instead of plugin-reddit

Commit:
Fix: fix starknet plugin by replacing walletProvider with portfolio provider

timmyg commented:
@Freytes  why did this get closed?

Commit:
20250110 tts

AIFlowML committed:
feat: Add Akash Network plugin with autonomous deployment capabilities

Hugo-SEQUIER committed:
New plugin version + ReadMe

Hugo-SEQUIER committed:
Merge branch 'develop' into main

mameikagou committed:
fix: rm unused imports

AIFlowML committed:
Merge upstream/develop and resolve conflicts

AIFlowML committed:
adding all the mising environments for Akash

web3gh committed:
fix sequence of configuration for vision model

tcm390 committed:
Merge branch 'develop' into main

tcm390 committed:
Merge pull request #2112 from mameikagou/main

fix: rm unused imports

wtfsayo committed:
Merge branch 'develop' into revert-#1808

tcm390 committed:
Merge pull request #2106 from elizaOS/revert-#1808

chore: revert #1808 + add missing import for coin gecko plugin

mameikagou committed:
Merge pull request #1 from elizaOS/develop

Develop

Commit:
Add Verified Inference docs

Hugo-SEQUIER committed:
Merge branch 'develop' into main

mameikagou committed:
fix: SttTtsPlugin incorrect interface implementation

tcm390 committed:
use model settings for image and remove duplicated files

tcm390 committed:
remove duplicated files

tcm390 committed:
remove duplicated files

Lukapetro committed:
fix: remove deprecated generateobject, templates

ag-wnl committed:
Merge pull request #2117 from mameikagou/develop

Add name and description to SttTtsPlugin

Lukapetro committed:
fix: readme.md docs

Lukapetro committed:
chore: resolve merge conflicts with develop

tcm390 committed:
handel infera provider

tcm390 committed:
add infera model type

tcm390 committed:
add infera provider

tcm390 committed:
Merge branch 'develop' into tcm-model-fix

shakkernerd committed:
Merge pull request #2118 from elizaOS/tcm-model-fix

fix: Apply model settings for images and remove duplicate files

wtfsayo committed:
Merge branch 'develop' into feat/coingecko-api-integration

tsubasakong committed:
Merge branch 'develop' into develop

tcm390 committed:
Merge branch 'develop' into main

tcm390 committed:
fetch user settings from runtime

wtfsayo committed:
update env example file

brauliolomeli committed:
add support for Mistral model in text generation

wtfsayo committed:
Merge pull request #2068 from Lukapetro/feat/coingecko-api-integration

feat: pro api support, trending coins api

wtfsayo committed:
Merge branch 'develop' into feature/akash-network-plugin-01

tsubasakong committed:
Merge branch 'develop' into develop

wtfsayo committed:
Merge branch 'develop' into supabase-updates

Hugo-SEQUIER committed:
Merge branch 'develop' into main

brauliolomeli committed:
refactor Mistral model settings structure for improved clarity and consistency

mcp97 committed:
Merge branch 'develop' into add_gemini_to_image_recognition

mcp97 committed:
Merge pull request #2099 from web3gh/add_gemini_to_image_recognition

Add gemini to image vision

mcp97 committed:
Merge branch 'develop' into feature/akash-network-plugin-01

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
chore: pnpm lock file check on PRs into main

odilitime committed:
Merge branch 'main' into plugin_import_fixes

shakkernerd committed:
Merge pull request #1708 from Hugo-SEQUIER/main

feat: Add Irys plugin

shakkernerd committed:
Merge branch 'develop' into feature/akash-network-plugin-01

odilitime committed:
Merge branch 'develop' into plugin_import_fixes

odilitime committed:
Merge branch 'develop' into develop

odilitime committed:
Merge pull request #2095 from treppers/plugin_import_fixes

fix: Fix plugin loading from a character.json file

odilitime committed:
Merge pull request #2093 from tsubasakong/develop

feat: add Heurist embedding model

tcm390 committed:
move truncate to core

odilitime committed:
Merge branch 'develop' into develop

tcm390 committed:
remove unuse parameter

odilitime committed:
Merge pull request #2103 from rferrari/develop

fix: clientConfig.telegram.isPartOfTeam misstype

wtfsayo committed:
Merge branch 'develop' into develop

odilitime committed:
Merge pull request #2125 from maciejwitowski/develop

docs: Add Verified Inference docs

odilitime committed:
Merge pull request #2091 from prasadabhishek/patch-2

fix: Update clients.md to fix package name

autonome-ai committed:
feat: support autonome platform (#2121)

* feat: support autonome platform

* Update packages/plugin-autonome/src/actions/launchAgent.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* fix typo

* Update packages/plugin-autonome/src/actions/launchAgent.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

---------

Co-authored-by: Sayo <hi@sayo.wtf>
Co-authored-by: Odilitime <janesmith@airmail.cc>
Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

odilitime committed:
Merge branch 'develop' into plugin

odilitime committed:
typo fix: close object

odilitime committed:
update lockfile

wtfsayo committed:
Merge branch 'develop' into pr/2111

tcm390 committed:
prevent  repeated login

tcm390 committed:
clean code

tcm390 committed:
modify comment

tcm390 committed:
return value

wtfsayo committed:
fix syntax issue

denizekiz committed:
monilpat suggested commentedout code blocks removed

denizekiz committed:
Merge branch 'develop' into interactionimage

pgoos committed:
run integration tests on the PR branch

tcm390 committed:
get settings from runtime

tcm390 committed:
Merge branch 'develop' into tcm-fix-plugin-twitter

wtfsayo committed:
optimistic fix

pgoos committed:
Merge branch 'develop' into piotr/coinbase-plugin-tests

wtfsayo committed:
test integrationTests

wtfsayo committed:
remove alpha

mcp97 committed:
Merge pull request #2131 from elizaOS/fix/broken-autonome

fix: syntax issue on autonome plugin and lock file update

mcp97 committed:
Merge branch 'develop' into piotr/coinbase-plugin-tests

tcm390 committed:
Merge branch 'develop' into tcm-fix-plugin-twitter

wtfsayo committed:
Merge branch 'develop' into feature/akash-network-plugin-01

mcp97 committed:
Merge pull request #1767 from Sifchain/piotr/coinbase-plugin-tests

test: Integration Tests Enhancement and Coinbase Commerce Integration

wtfsayo committed:
Merge branch 'develop' into feature/akash-network-plugin-01

shakkernerd committed:
Merge branch 'develop' into tcm-fix-plugin-twitter

shakkernerd committed:
Merge pull request #2129 from elizaOS/tcm-fix-plugin-twitter

fix: prevent repeated login by reusing client-twitter session

wtfsayo committed:
Merge branch 'develop' into feature/akash-network-plugin-01

denizekiz committed:
Merge branch 'develop' into interactionimage

odilitime committed:
lint fixes

odilitime committed:
lint fixes

odilitime committed:
Merge branch 'develop' into odi-fix

odilitime committed:
processAtions can't be awaited in non-async function

odilitime committed:
processAtions can't be awaited in non-async function

odilitime committed:
revert GoPlusType so it can work with switch statement

odilitime committed:
revert GoPlusType so it can work with switch statement

odilitime committed:
fix: update lockfile and fix lint findings (#2128)

* typo fix: close object

* update lockfile

wtfsayo committed:
Update packages/plugin-akash/src/actions/getDeploymentApi.ts

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

wtfsayo committed:
Merge pull request #2111 from AIFlowML/feature/akash-network-plugin-01

feat: Add Akash Network plugin with autonomous deployment capabilities

odilitime committed:
Merge branch 'develop' into plugin

brauliolomeli committed:
add support for Mistral model in text generation and configuration

odilitime committed:
Merge branch 'develop' of https://github.com/elizaos/eliza into develop

wtfsayo committed:
Merge pull request #2101 from pranjallyad/plugin

feat: Lens Network Plugin

brauliolomeli committed:
add Mistral model configuration to environment example and documentation

odilitime committed:
Merge branch 'develop' into pr1531

odilitime committed:
Merge branch 'develop' of https://github.com/elizaos/eliza into pr1531

brauliolomeli committed:
Merge remote-tracking branch 'origin/develop' into mistral-text-generation

odilitime committed:
mark unused var for lint

odilitime committed:
bump lock

odilitime committed:
pnpm audit fix

odilitime committed:
Merge branch 'pr1531' of https://github.com/odilitime/eliza into pr1531

tudorpintea999 committed:
Update evaluators.md

tudorpintea999 committed:
Update actions.md

tudorpintea999 committed:
Update characterfile.md

tcm390 committed:
Merge branch 'develop' into text-fixes

tcm390 committed:
Merge pull request #2138 from tudorpintea999/text-fixes

Docs improvements

earlyvibz committed:
feat: plugin-hyperliquid

earlyvibz committed:
chore: sync with develop and fix conflicts

tcm390 committed:
fix export name

tcm390 committed:
duplicated import

shakkernerd committed:
Merge pull request #2142 from elizaOS/tcm-fix-lens-exportname

fix: lens export name and duplicate imports

odilitime committed:
Merge branch 'develop' into pr1531

mcp97 committed:
elizaLogger

mcp97 committed:
Merge branch 'develop' into supabase-updates

mcp97 committed:
Merge pull request #2100 from antman1p/supabase-updates

Fix: Supabase updates

tcm390 committed:
Merge branch 'develop' into pr1531

odilitime committed:
Merge pull request #2135 from odilitime/pr1531

refactor: Optimize memory fetching by moving sorting and slicing to DB (PR #1531 remake)

odilitime committed:
Merge branch 'develop' into feature/cloudflare-ai-gateway-clean

odilitime committed:
clean up anthropic

odilitime committed:
Merge pull request #821 from w3-bounty/feature/cloudflare-ai-gateway-clean

feat: Add Cloudflare AI Gateway support

odilitime committed:
Merge branch 'develop' into feature/plugin-hyperliquid

tcm390 committed:
Merge branch 'develop' into main

odilitime committed:
Merge pull request #2088 from rRogick/main

docs: add readme portuguese version in docs

AIFlowML committed:
All eslint errros fixed and re-tested all the actions again

odilitime committed:
Merge branch 'develop' into fix-akash-eslint

odilitime committed:
Merge pull request #2143 from AIFlowML/fix-akash-eslint

fix: eslint-fix-full-test-again

tcm390 committed:
Merge pull request #2029 from Jonatan-Chaverri/fix_starknet_plugin

Fix: fix starknet plugin by replacing walletProvider with portfolio provider

denizekiz committed:
Merge branch 'develop' into interactionimage

wtfsayo committed:
Merge branch 'develop' into feature/plugin-hyperliquid

Commit:
fix: correct SUI/USD price calculation

wtfsayo committed:
Merge pull request #1775 from denizekiz/interactionimage

feat: Image descriptions into interaction.ts

wtfsayo committed:
Merge branch 'develop' into feature/plugin-hyperliquid

wtfsayo committed:
Merge pull request #2141 from earlyvibz/feature/plugin-hyperliquid

feat: plugin-hyperliquid

rellfy committed:
chore: merge upstream

rellfy committed:
Merge branch 'develop' into main

wtfsayo committed:
Merge pull request #2137 from GravitonINC/mistral-text-generation

feat: Add Mistral AI as new model provider

ag-wnl committed:
Merge pull request #2045 from asterai-io/main

feat: Implement asterai plugin

wtfsayo committed:
Merge branch 'develop' into solana_transaction_more_lenient

wtfsayo committed:
Merge branch 'develop' into pr/1582

wtfsayo committed:
resolve conflicts

wtfsayo committed:
Merge pull request #1582 from peterjah/massa_plugin

feat: add massa-plugin

wtfsayo committed:
Merge branch 'develop' into pr/2083

wtfsayo committed:
Update .env.example

wtfsayo committed:
Merge pull request #2083 from 0xalank/quai-integration

feat: add Quai integration

xudean committed:
Merge remote-tracking branch 'upstream/develop' into develop

# Conflicts:
#	pnpm-lock.yaml

Commit:
Update decimals

wtfsayo committed:
Merge branch 'develop' into pr/2067

wtfsayo committed:
Update pnpm-lock.yaml

wtfsayo committed:
Update types.ts

wtfsayo committed:
fix env examples

wtfsayo committed:
Merge branch 'main' of https://github.com/daizhengxue/eliza into pr/2067

wtfsayo committed:
Update .env.example

wtfsayo committed:
Update models.ts

wtfsayo committed:
Merge pull request #2067 from daizhengxue/main

feat: Add DeepSeek AI provider support to Eliza

ChainRex committed:
Merge branch 'develop' into fix-plugin-sui

xudean committed:
Merge remote-tracking branch 'upstream/develop' into develop

# Conflicts:
#	pnpm-lock.yaml

xudean committed:
resolve conversions problems

wtfsayo committed:
Merge pull request #2086 from primus-labs/develop

feat: Primus zkTLS plugin to fully verify agent activities

ShreyGanatra committed:
inheritance of character from parent using extends key

AnonJon committed:
add github to client enumerations

tcm390 committed:
fix join voice channel

odilitime committed:
bump lock

odilitime committed:
merge, fix conflicts

antman1p committed:
Merge branch 'develop' into patch-3

tcm390 committed:
add missing function and parameters

odilitime committed:
Merge pull request #2160 from elizaOS/tcm-join-voice

fix: fix client-discord join voice action

mcp97 committed:
Merge branch 'develop' into fix/add-github-client-enum

mcp97 committed:
Merge pull request #2157 from AnonJon/fix/add-github-client-enum

add github to client enumerations

mcp97 committed:
Merge branch 'develop' into inheritance

mcp97 committed:
Merge pull request #2159 from bark-ruffalo/inheritance

inheritance of character from parent using extends key

odilitime committed:
Merge branch 'odi-fix' of https://github.com/odilitime/eliza into odi-fix

mcp97 committed:
Merge pull request #2150 from ChainRex/fix-plugin-sui

fix: correct SUI/USD price calculation

odilitime committed:
Merge branch 'develop' into odi-fix

mcp97 committed:
Merge branch 'develop' into solana_transaction_more_lenient

mcp97 committed:
Merge pull request #2053 from zhourunlai/solana_transaction_more_lenient

feat: solana transaction more lenient

mcp97 committed:
Merge branch 'develop' into add-uuid-tests-and-fix-version

mcp97 committed:
Update uuid.ts

update so just adding test

mcp97 committed:
Merge pull request #1362 from shlokkhemani/add-uuid-tests-and-fix-version

chore: Add UUID tests and fix version 5 bits

mcp97 committed:
Merge branch 'develop' into readme-git-command

mcp97 committed:
Merge pull request #1705 from velvet-shark/readme-git-command

Update git command for checking latest release

mcp97 committed:
Merge branch 'develop' into patch-3

mcp97 committed:
Merge pull request #2032 from antman1p/patch-3

chore: add conditionals for supabase to agent directory

odilitime committed:
chore: bump develop lockfile (#2166)

* typo fix: close object

* update lockfile

* lint fixes

* processAtions can't be awaited in non-async function

* revert GoPlusType so it can work with switch statement

* bump lock

* merge, fix conflicts

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #2110 from tomguluson92/main

feat: TTS(Text2Speech) with over 15 languages support!

ai16z-demirix committed:
test: moving uuid.tests to correct __tests__ directory

mcp97 committed:
Merge pull request #2168 from ai16z-demirix/test/structure

ai16z-demirix committed:
feature: adding tests for coinbase plugin to __tests__ directory

mcp97 committed:
Merge branch 'develop' into tests/coinbase-plugin

mcp97 committed:
Merge pull request #2170 from ai16z-demirix/tests/coinbase-plugin

tests: coinbase plugin - adding tests for coinbase plugin

Jonatan-Chaverri committed:
Fix: replace invalid toghether ai medium model

rferrari committed:
fix: insert missing langdetect on package.json

AAAkater committed:
🎈 perf(vscode): Set file nesting for md and DockerFile

nhodges committed:
fix(client-twitter): add mention deduplication utility

Currently, the Grok LLM is generating duplicate mentions at the beginning of responses. This adds a simple safeguard to deduplicate mentions before sending the tweet.

nhodges committed:
Merge branch 'develop' into patch-1

odilitime committed:
Merge pull request #2173 from Jonatan-Chaverri/fix_together_model

Fix: replace invalid toghether ai medium model

odilitime committed:
Merge branch 'develop' into develop

odilitime committed:
Merge pull request #2175 from rferrari/develop

fix: insert missing langdetect on plugin-tts package.json

odilitime committed:
Merge branch 'develop' into patch-1

odilitime committed:
Merge pull request #2178 from nhodges/patch-1

fix(client-twitter): add mention deduplication utility

odilitime committed:
Merge pull request #2177 from AAAkater/main

feat: 🎈 perf(vscode): Set file nesting for md and DockerFile

odilitime committed:
Merge branch 'develop' of https://github.com/ai16z/eliza into odi-fix

Riroaki committed:
Fix deepseek support in getTokenForProvider (#2179)

Co-authored-by: Odilitime <janesmith@airmail.cc>

jonathanykh committed:
fix: remove problematic redundant uuid conversion and add api input param validations to api server (#2051)

* fix: remove problematic redundant uuid conversion and add api input param validations to api server

* style: use object property shorthand for roomId

* chore: update pnpm-lock.yaml

---------

Co-authored-by: Monil Patel <monilpat@gmail.com>
Co-authored-by: Odilitime <janesmith@airmail.cc>

odilitime committed:
Merge branch 'develop' of https://github.com/ai16z/eliza into odi-fix

zoe27 committed:
fix: fix the chat stuck in infinite loop (#1755)

* fix the chat stuck in infinite loop

* perfect the PR and keep the temprature and wordsToPunish in the generate response

* Update README.md

* Trigger CI checks

---------

Co-authored-by: Odilitime <janesmith@airmail.cc>

odilitime committed:
Merge branch 'develop' of https://github.com/ai16z/eliza into odi-fix

web3gh committed:
fix: Koloxarto/fix ragknowledge for postgres (#2153)

* fix formatting out of the way

* fix postgress chunk uuid handling for ragKnowledge

---------

Co-authored-by: Odilitime <janesmith@airmail.cc>

odilitime committed:
convert imageDescriptionsArray from let to const per lint

odilitime committed:
remove duplicate TOGETHER in case, lint/unused var

odilitime committed:
bump eslint so it doesn't crash

odilitime committed:
comment out unused AkashMessage interface

odilitime committed:
clean up unused var in catch

odilitime committed:
bump

odilitime committed:
Merge branch 'odi-fix' of https://github.com/odilitime/eliza into odi-fix

odilitime committed:
Merge branch 'develop' of https://github.com/ai16z/eliza into odi-fix

odilitime committed:
Merge branch 'odi-fix' of https://github.com/ai16z/eliza into odi-fix

odilitime committed:
bump version to 0,1,8

odilitime committed:
chore: lint and fix pass on develop (#2180)

* typo fix: close object

* update lockfile

* lint fixes

* processAtions can't be awaited in non-async function

* revert GoPlusType so it can work with switch statement

* lint fixes

* processAtions can't be awaited in non-async function

* revert GoPlusType so it can work with switch statement

* bump lock

* merge, fix conflicts

* convert imageDescriptionsArray from let to const per lint

* remove duplicate TOGETHER in case, lint/unused var

* bump eslint so it doesn't crash

* comment out unused AkashMessage interface

* clean up unused var in catch

* bump

odilitime committed:
Merge pull request #2171 from elizaOS/develop

chore: Prep v0.1.8 (dev => main)

ali-moha committed:
Add Persian README File

odilitime committed:
fix path

odilitime committed:
fix quai deps

odilitime committed:
Merge branch 'develop' into odi-fix

odilitime committed:
fix json format typo

daizhengxue committed:
Update types.ts

odilitime committed:
Merge branch 'odi-fix' of https://github.com/odilitime/eliza into odi-fix

odilitime committed:
Merge pull request #2184 from odilitime/odi-fix

fix: release 0.1.8 fixes

odilitime committed:
Merge branch 'develop' into add_persian_readme

odilitime committed:
Merge pull request #2182 from ali-moha/add_persian_readme

docs: Add Persian README File

odilitime committed:
Merge pull request #2187 from daizhengxue/patch-1

fix: Missing LETZAI model

web3gh committed:
fix postgres adapter migration extension creation which already exists at this point (#2188)

nhodges committed:
fix(client-twitter): clean up mention deduplication (#2185)

Co-authored-by: Odilitime <janesmith@airmail.cc>

AIFlowML committed:
feat(security): Implement comprehensive file upload security measures - Add FileSecurityValidator, file type restrictions, size limits, path traversal prevention, enhanced logging and security documentation (#1753) (#1806)

Co-authored-by: Odilitime <janesmith@airmail.cc>

odilitime committed:
bump version (#2193)

* chore: lint and fix pass on develop (#2180)

* typo fix: close object

* update lockfile

* lint fixes

* processAtions can't be awaited in non-async function

* revert GoPlusType so it can work with switch statement

* lint fixes

* processAtions can't be awaited in non-async function

* revert GoPlusType so it can work with switch statement

* bump lock

* merge, fix conflicts

* convert imageDescriptionsArray from let to const per lint

* remove duplicate TOGETHER in case, lint/unused var

* bump eslint so it doesn't crash

* comment out unused AkashMessage interface

* clean up unused var in catch

* bump

* Add Persian README File

* fix path

* fix quai deps

* fix json format typo

* Update types.ts

* fix postgres adapter migration extension creation which already exists at this point (#2188)

* fix(client-twitter): clean up mention deduplication (#2185)

Co-authored-by: Odilitime <janesmith@airmail.cc>

* feat(security): Implement comprehensive file upload security measures - Add FileSecurityValidator, file type restrictions, size limits, path traversal prevention, enhanced logging and security documentation (#1753) (#1806)

Co-authored-by: Odilitime <janesmith@airmail.cc>

* bump version

---------

Co-authored-by: Ali <amohammadzadeh@interrodata.com>
Co-authored-by: Masterdai <daizhengxue828@gmail.com>
Co-authored-by: koloxarto <106516707+web3gh@users.noreply.github.com>
Co-authored-by: Nuri Hodges <nhodges@users.noreply.github.com>
Co-authored-by: AIFlow_ML <mlstudio@aiflow.ml>

odilitime committed:
Merge pull request #2194 from elizaOS/develop

chore: 0.1.8.build.1 (dev => main)

tomguluson92 committed:
Add 0G to the blockchain sector

tcm390 committed:
Merge pull request #2204 from tomguluson92/main

feats: Add 0G to the blockchain sector (diagram update)

bodhi-crypo committed:
fix(plugin-depin): sentientAI description (#2668)
